### DISCUSSION TITLE
Add preview for markdown

### DISCUSSION THREAD

**Discussion by emilyacook** [mattermost/community#65] [Enhancements]
A preview feature for messages would be nice since we're using Markdown. Right now I just send a lot of messages to myself

**Comment by chenilim** [comment]
Yes! I do this all the time - even drafting the post in a separate Markdown app before pasting it in.

**Comment by esethna** [comment]
There's a "preview" feature available today actually: @emilyacook @chenilim 

![image](https://user-images.githubusercontent.com/14333569/123169074-aca45600-d42d-11eb-98fc-c883348bea8b.png)


But the end game here is the advanced text editor that has a preview mode: https://mattermost.atlassian.net/wiki/spaces/ATE/overview?homepageId=1015152748

![image](https://user-images.githubusercontent.com/14333569/123169240-dd848b00-d42d-11eb-82ea-b22fb022961f.png)

**Reply by chenilim** [reply]
Cool! Thanks for pointing that out! Just enabled it!

**Reply by Willyfrog** [reply]
I thought this was enabled by default! Shouldn't we move it out of pre-release?

**Reply by esethna** [reply]
Yes we should! Or just turn it on by default until we have the advanced editor

**Reply by zefhemel** [reply]
Funny thing is that I enabled this ages ago, but only now noticed that this actually adds a (rather subtle) "Preview" link to under the text box to actually use it. I think the tweak of even just having that "eye" in text-area icon as shown for the advanced editor would make this feature more discoverable. And yes, of course let's also enable it by default :D

**Comment by chetanyakan** [comment]
This can be made more known to our users if we implement these tips and tricks: https://github.com/mattermost/community/discussions/92

**Reply by Willyfrog** [reply]
It would help to discover it. But I still prefer to have it enabled by default and have the tips and tricks.

**Reply by esethna** [reply]
+1, functionality that benefits everyone should just be default on

**Reply by chetanyakan** [reply]
Actually, the preview renders wierdly (i.e. not the same as what would appear in the actual post) in some cases like when single-line code blocks are present.

Maybe that's why the feature is marked experimental and is disabled by default.
